### PR TITLE
Okta 430932: missing password requirements

### PR DESCRIPTION
**Pull request by lesterchoi-okta**
## Description:

Need some feedback on this fix. AFAICT, there are 3 options to address the a11y issue outlined in <https://oktainc.atlassian.net/browse/OKTA-430932>. The satisfied/unsatisfied status of requirements should be audible for screen-reader users to fulfill (by adjusting their input).

* Option A: add `role="status"` or `role="alert"` to `<p>` tag
* Option B: add "sr-only" css class to prepended `<span>` with localized string indicating "unsatisfied" or similar.
* Option C: show password hints before input field (see https://incl.ca/show-hide-password-accessibility-and-password-hints-tutorial/)

Option A is best because it avoids the need for updated localizations and achieves the same effect as Option B but in a more semantic/standardized way, but I can't seem to produce the expected speech in NVDA on Windows.

## PR Checklist

- [ ] Have you verified the basic functionality for this change?

- [ ] Added unit tests?

- [ ] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

### Issue:

- [OKTA-430932](https://oktainc.atlassian.net/browse/OKTA-430932)

### CODE REVIEW

File: playground/mocks/config/responseConfig.js [hunk 1]
```diff
@@ -835,6 +835,12 @@ const secondaryEmail = {
...
 
+const selfServiceRegistration = {
+  '/api/v1/registration/form': [
+    'registration-form'
+  ]
+};
+
 module.exports = {
-  mocks: idx,
+  mocks: selfServiceRegistration,
```
Comment (shawyu-okta) [inline playground/mocks/config/responseConfig.js:845]: May need to revert this change. I assume you edited for local testing but I think `idx` mocks have been the default scenario configured.

File: src/util/RegistrationFormFactory.js [hunk 2]
```diff
@@ -113,16 +113,31 @@ const checkSubSchemas = function(fieldName, model, subSchemas, showError) {
...
       if (showError) {
         ele.find('p span').removeClass('confirm-16');
         ele.find('p span').addClass('error error-16-small');
         ele.addClass('subschema-error subschema-unsatisfied');
+
+        // set role="status" on <p>
+        ele.find('p').attr('role', 'status');
+        
+        // ele.find('p .sr-only').length === 0 && ele.find('p').prepend('<span class="sr-only" >Error</span>');
+        // ele.find('.sr-only').length === 0 && ele.prepend('<span class="sr-only">Error</span>');
```
Comment (shawyu-okta) [inline src/util/RegistrationFormFactory.js:140]: Flagging for removal if the commented code is unused in the final PR

File: src/util/RegistrationFormFactory.js [hunk 3]
```diff
@@ -113,16 +113,31 @@ const checkSubSchemas = function(fieldName, model, subSchemas, showError) {
     } else {
       ele.children('p').addClass('default-schema');
     }
+
+    // clear role="status" on <p>
+    ele.children('p').removeAttr('role');
+
+    // reset errors
```
Comment (shawyu-okta) [inline src/util/RegistrationFormFactory.js:120]: 💯  for some comment annotations in this code

File: webpack.playground.config.js [hunk 4]
```diff
@@ -63,7 +65,9 @@ module.exports = {
     ],
     historyApiFallback: true,
     headers: {
-      'Content-Security-Policy': `script-src http://localhost:${DEV_SERVER_PORT}`
+      'Content-Security-Policy': [
+        `script-src http://${HOST}:${DEV_SERVER_PORT}`
+      ].join(';')
```
Comment (shawyu-okta) [inline webpack.playground.config.js:70]: Does changing this string construction make a difference or is it just for future devs who need to add more CSP items?

File: src/util/RegistrationFormFactory.js [hunk 5]
```diff
@@ -113,16 +113,31 @@ const checkSubSchemas = function(fieldName, model, subSchemas, showError) {
     } else {
       ele.children('p').addClass('default-schema');
     }
+
+    // clear role="status" on <p>
+    ele.children('p').removeAttr('role');
```
Comment (shawyu-okta) [inline src/util/RegistrationFormFactory.js:118]: Is there a way to remove the role in the DOM by default rather than having to do it in JS?

File: playground/mocks/config/responseConfig.js (hunk 1 above)
Comment (lkchoi) [reply playground/mocks/config/responseConfig.js:845]: Yeah that won't get merged. I usually leave a note to self DONKTMERGE. Just forgot this time

File: src/util/RegistrationFormFactory.js (hunk 2 above)
Comment (lkchoi) [reply src/util/RegistrationFormFactory.js:140]: Those are the other options. B and C left them for now to easily try them out. I'll remove the code we don't use

File: src/util/RegistrationFormFactory.js [hunk 6]
```diff
@@ -113,16 +113,35 @@ const checkSubSchemas = function(fieldName, model, subSchemas, showError) {
     } else {
       ele.children('p').addClass('default-schema');
     }
+
+    // clear aria role and live-region for re-validation
+    ele.children('p')
+      .removeAttr('role')
+      .removeAttr('aria-live');
```
Comment (shawyu-okta) [inline src/util/RegistrationFormFactory.js:120]: Possibly worth seeing if this can live inside the `// validate` block but I think it's fine here too.

Comment (lesterchoi-okta) [reply src/util/RegistrationFormFactory.js:120]: true. i think it's safer outside so we know it's back to a "default" state before applying the role and aria-live attributes again in the failed condition

File: playground/mocks/spec-okta-api/api/v1/registration/form.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
+/* eslint-disable @okta/okta/no-unlocalized-text */
+const templateHelper = require('../../../../config/templateHelper');
+
+module.exports = [
+  templateHelper({
+    path: '/api/v1/registration/form',
+    method: 'GET',
+    template: JSON.stringify({
```
Comment (haishengwu-okta) [inline playground/mocks/spec-okta-api/api/v1/registration/form.js:8]: Hi @lesterchoi-okta , I assume specify `template` isn't required (or actually be abundant) ?

Comment (lesterchoi-okta) [reply playground/mocks/spec-okta-api/api/v1/registration/form.js:8]: @haishengwu-okta yes, think you're right. i'm going to add a cleanup task and follow up right after release
